File: src/TypeScript/TypeScriptLibrary/Library/CategoryArrow.ts
import { ICategoryArrow } from "./Interfaces/ICategoryArrow";
import { ICategoryObject } from "./Interfaces/ICategoryObject";
import { IDesktop } from "./Interfaces/IDesktop";
import { IObject } from "./Interfaces/IObject";

export class CategoryArrow implements ICategoryArrow, IObject
{
    constructor(desktop: IDesktop, name: string)
    {
        this.desktop = desktop;
        this.name = name;
        desktop.addCategoryArrow(this);
    }
    getName(): string {
        throw new Error("Method not implemented.");
    }

    getCategoryArrowName(): string
    {
        return this.name;
    }

    getClassName(): string {
        return this.typeName;
    }
    imlplementsType(type: string): boolean {
        return this.types.indexOf(type) >= 0;
    }

    protected name !: string;

    protected desktop !: IDesktop;

    getDesktop(): IDesktop {
        return this.desktop;
    }

    source!: ICategoryObject;

    target!: ICategoryObject;

    protected typeName: string = "CategoryArrow";

    protected types: string[] = ["ICategoryArrow", "CategoryArrow"];


    getSource(): ICategoryObject {
        return this.source;
    }
    getTagret(): ICategoryObject {
        return this.target;
    }
    setSource(source: ICategoryObject): void {
        this.source = source;
    }
    setTarget(target: ICategoryObject): void {
        this.target = target;
    }
}

File: Java_Generated/Java/untitled/src/conversion/category_theory/CategoryArrow.java
package conversion.category_theory;

import conversion.category_theory.interfaces.ICategoryArrow;
import conversion.category_theory.interfaces.ICategoryObject;
import conversion.category_theory.interfaces.IDesktop;

public class CategoryArrow implements ICategoryArrow {

    protected  ICategoryObject source;

    protected  ICategoryObject target;

    protected String name;

    protected  IDesktop desktop;

    public CategoryArrow(String name, IDesktop desktop)
    {
        this.name = name;
        this.desktop = desktop;
        if (desktop != null)
        {
            desktop.addCategoryArrow(this);
        }

    }

    /**
     * @return
     */
    @Override
    public ICategoryObject getSource() {
        return source;
    }

    /**
     * @return
     */
    @Override
    public ICategoryObject getTagret() {
        return target;
    }

    /**
     * @param source
     */
    @Override
    public void setSource(ICategoryObject source) {
this.source = source;
    }

    /**
     * @param target
     */
    @Override
    public void setTarget(ICategoryObject target) {

    }

    /**
     * @return
     */
    @Override
    public String getArrowName() {
        return name;
    }

    /**
     * @return
     */
    @Override
    public IDesktop getDesktop() {
        return desktop;
    }
}

